lib/xfrm/sa.c

```diff
@@ -1006,6 +1010,11 @@ int xfrmnl_sa_parse(struct nlmsghdr *n, struct xfrmnl_sa **result)
 		sa->ce_mask |= XFRM_SA_ATTR_OFFLOAD_DEV;
 	}
 
+	if (tb[XFRMA_IF_ID]) {
+		sa->if_id      =   *(uint32_t*)nla_data(tb[XFRMA_IF_ID]);
+		sa->ce_mask |= XFRM_SA_ATTR_IF_ID;
```

Comment:
you must add an entry to `xfrm_sa_policy` to ensure that the provided data is large enough to hold a uint32.

Reply:
Done.